testing/simulated/simulated_test.go

```diff
@@ -0,0 +1,261 @@
+func (s *SimulatedSuite) TearDownTest() {
+	if err := s.ElHandle.Close(); err != nil {
+		s.T().Error("Error closing EL handle:", err)
+	}
+	s.CancelFunc()
+}
+
+// initializeChain sets up the chain using the genesis file.
+func (s *SimulatedSuite) initializeChain() {
+	// Load the genesis state.
+	appGenesis, err := genutiltypes.AppGenesisFromFile(s.HomeDir + "/config/genesis.json")
+	s.Require().NoError(err)
+
+	// Initialize the chain.
+	initResp, err := s.SimComet.Comet.InitChain(s.Ctx, &types.InitChainRequest{
+		ChainId:       simulated.TestnetBeaconChainID,
+		AppStateBytes: appGenesis.AppState,
+	})
+	s.Require().NoError(err)
+	s.Require().Len(initResp.Validators, 1, "Expected 1 validator")
+
+	// Verify that the deposit store contains the expected deposits.
+	deposits, err := s.TestNode.StorageBackend.DepositStore().GetDepositsByIndex(
+		s.Ctx,
+		constants.FirstDepositIndex,
+		constants.FirstDepositIndex+s.TestNode.ChainSpec.MaxDepositsPerBlock(),
+	)
+	s.Require().NoError(err)
+	s.Require().Len(deposits, 1, "Expected 1 deposit")
+}
+
+// TestFullLifecycle_ValidBlock_IsSuccessful tests that a valid block proposal is processed, finalized, and committed.
+func (s *SimulatedSuite) TestFullLifecycle_ValidBlock_IsSuccessful() {
+	const blockHeight = 1
+
+	// Initialize the chain state.
+	s.initializeChain()
+
+	// Retrieve the BLS signer and proposer address.
+	blsSigner := simulated.GetBlsSigner(s.HomeDir)
+	pubkey, err := blsSigner.GetPubKey()
+	s.Require().NoError(err)
+
+	// Prepare a block proposal.
+	proposalTime := time.Now()
+	proposal, err := s.SimComet.Comet.PrepareProposal(s.Ctx, &types.PrepareProposalRequest{
+		Height:          blockHeight,
+		Time:            proposalTime,
+		ProposerAddress: pubkey.Address(),
+	})
+	s.Require().NoError(err)
+	s.Require().NotEmpty(proposal)
+
+	// Process the proposal.
+	processResp, err := s.SimComet.Comet.ProcessProposal(s.Ctx, &types.ProcessProposalRequest{
+		Txs:             proposal.Txs,
+		Height:          blockHeight,
+		ProposerAddress: pubkey.Address(),
+		Time:            proposalTime,
+	})
+	s.Require().NoError(err)
+	s.Require().Equal(types.PROCESS_PROPOSAL_STATUS_ACCEPT, processResp.Status)
+
+	// Finalize the block.
+	finalizeResp, err := s.SimComet.Comet.FinalizeBlock(s.Ctx, &types.FinalizeBlockRequest{
+		Txs:             proposal.Txs,
+		Height:          blockHeight,
+		ProposerAddress: pubkey.Address(),
+	})
+	s.Require().NoError(err)
+	s.Require().NotEmpty(finalizeResp)
+
+	// Commit the block.
+	_, err = s.SimComet.Comet.Commit(s.Ctx, &types.CommitRequest{})
+	s.Require().NoError(err)
+
+	// Validate post-commit state.
+	queryCtx, err := s.SimComet.CreateQueryContext(blockHeight, false)
+	s.Require().NoError(err)
+
+	stateDB := s.TestNode.StorageBackend.StateFromContext(queryCtx)
+	slot, err := stateDB.GetSlot()
+	s.Require().NoError(err)
+	s.Require().Equal(mathpkg.U64(blockHeight), slot)
+
+	stateHeader, err := stateDB.GetLatestBlockHeader()
+	s.Require().NoError(err)
```

Comment:
q: can't we use some APIs to retrieve the header, similarly to what we do in e2e tests?